datalad/distribution/install.py

```diff
@@ -538,6 +539,8 @@ def __call__(
 
             # We possibly need to consider /.git URL
             candidate_source_urls = [source_url]
+            # TODO: isn't this a duplicate of above logic/implementation
+            # in _install_subds_from_flexible_source????
```

Comment:
Yes, it is. There is some RF'ing to be done regarding `_install_subds_from_flexible_source`, `GitRepo.clone()` and this one. They all try to clone from possibly multiple URLs.